servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/Exceptions.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2021-2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.ConnectionRejectedException;
+import io.servicetalk.client.api.NoActiveHostException;
+import io.servicetalk.client.api.NoAvailableHostException;
+import io.servicetalk.concurrent.internal.ThrowableUtils;
+
+final class Exceptions {
+
+    static final class StacklessNoAvailableHostException extends NoAvailableHostException {
+        private static final long serialVersionUID = 5942960040738091793L;
+
+        private StacklessNoAvailableHostException(final String message) {
+            super(message);
+        }
+
+        @Override
+        public Throwable fillInStackTrace() {
+            return this;
+        }
+
+        public static StacklessNoAvailableHostException newInstance(String message, Class<?> clazz, String method) {
```

Comment:
Understand it was moved as-is. While you are here, please make all these `newInstance` methods pkg-private